operator/cilium_node.go

```diff
@@ -164,13 +203,38 @@ func startSynchronizingCiliumNodes(nodeManager allocator.NodeEventHandler, withK
 	go func() {
 		cache.WaitForCacheSync(wait.NeverStop, ciliumNodeInformer.HasSynced)
 		close(k8sCiliumNodesCacheSynced)
+		// Only handle events if syncHandler is not nil. If it is nil then
+		// there isn't any event handler set for CiliumNodes events.
+		if syncHandler != nil {
+			for processNextWorkItem(queue, syncHandler) {
+			}
+		}
 	}()
 
 	go ciliumNodeInformer.Run(wait.NeverStop)
 
 	return nil
 }
 
+// processNextWorkItem process all events from the workqueue.
+func processNextWorkItem(queue workqueue.RateLimitingInterface, syncHandler func(key string) error) bool {
+	key, quit := queue.Get()
+	if quit {
+		return false
+	}
+	defer queue.Done(key)
+
+	err := syncHandler(key.(string))
+	if err == nil {
+		// If err is nil we can forget it from the queue, if it is not nil
+		// the queue handler will retry to process this key until it succeeds.
+		queue.Forget(key)
+		return true
+	}
+
+	return true
```

Comment:
I recall @ldelossa mentioning some nuance where in some case (forgot what it was) that we need to make some extra call to `queue.Add()` or similar. It that addressed here? Louis, could you remind me what that was and validate whether this code already addressed it?

Reply:
Nice call @christarazi I've added that missing piece.